tests/sharingTest.js

```diff
@@ -627,10 +630,11 @@ describe('Main: Currently testing file/folder sharing,', function () {
               '.*\\/ocs\\/v1\\.php\\/apps\\/files_sharing\\/api\\/v1\\/shares$',
               '/ocs/v1.php/apps/files_sharing/api/v1/shares'
             ),
-            headers: validAuthHeaders
-            // TODO: uncomment this line once the issue is fixed
-            // https://github.com/pact-foundation/pact-js/issues/577
-            // body: 'shareType=0&shareWith=' + config.testUser + '&path=' + config.testFilesPath[i] + '&expireDate=' + config.expirationDate
+            headers: {
+              ...validAuthHeaders,
+              ...applicationFormUrlEncoded
+            },
+            body: 'shareType=0&shareWith=' + config.testUser + '&path=' + file + '&expireDate=' + config.expirationDate
```

Comment:
this function is now called through a `map` now and does not have a loop of its own, so I had to change the path